[EventGhost] - Enhancement - Log.py, Upgrades the debug logging

kdschlosser opened the pull request:
Changes the debug logging to find the name of the calling module and if it is a plugin it will create a new file specifically for that plugin. Provides alittle more organization to the debug logging. It also houses all of the log files into a folder called logs in %appdata%\EventGhost. 

It also fixes the issue of not having the debug header information added if the user has logging turned on from inside of the GUI. and I added the Windows version, Windows major, minor, build version, the start time of EG is all added to this header also. The header is added to each new file that is made or appended to one that is already existing. It closes the file after each write to it so it can be deleted without the need to shutdown EG.

If there happen to be any error during the load of EG this information will be written to the standard Log.txt file. and anything that is not a plugin that is writing log information is written to Core.txt in the logs folder.

kdschlosser committed:
Upgrades the debug logging

Changes the debug logging to find the name of the calling module and if it is a plugin it will create a new file specifically for that plugin. Provides alittle more organization to the debug logging.

topic2k commented on eg/Classes/Log.py:
```diff
@@ -34,11 +36,59 @@
+    'Windows Version: %s' % ' '.join(
+        list(platform.win32_ver())
+    ),
+    'Computer Name: %s' % platform.node(),
```
Do we really need the computer name for debugging?

topic2k commented on eg/Classes/Log.py:
```diff
@@ -34,11 +36,59 @@
+    ),
+    'Python Compiler: %s' % platform.python_compiler(),
+    'Python Build Date: %s' % (
+        platform.python_build()[1],
```
I'm missing the info that it's stackless python, so i would just use `sys.version`. It returns `'2.7.12 Stackless 3.1b3 060516 (default, Oct 31 2016, 18:42:08) [MSC v.1500 32 bit (Intel)]'
`

kdschlosser replied:
ok can do
I'll do it right now

kdschlosser replied:
do you like the way it works tho? I likle it alot better. it is a pain in the backside to sift through a debug log file when you are looking for something that is specific to a plugin. This is a nice way to handle it.

topic2k replied:
maybe if i would use EG more then i would find it more usefull :D but i will merge it anyway ;)

kdschlosser replied:
all done.

topic2k reviewed (commented):
I personaly don't like to have the log splitted, but that's  just me. Don't know what others say. I hope we get some more comments.

kdschlosser committed:
Fixes python version

Adds the Stackless version to the log header

kdschlosser committed:
Adds variable to eg.Config for turning on split logging

if split logging is to be used the config.py file needs to be edited and the line splitDebuggingLog = True needs to be added to it in order to turn on this feature.

added for printing the header information to the original Log output. 
removed the computer name from the header.

kdschlosser commented:
corrected the python version.
removed the computer name.

added the ability to turn on or off the slpit logging via adding a line to the config.py file

    splitDebuggingLog = True

made it so that each time the debugging log is turned on or off it will set a flag for printing the header to the original Log.txt

kdschlosser commented:
Oh and this will only split off the plugins and nothing else. I am not usually one for split logs either. But when i specifically am looking for output form a plugin i am working on it's an annoyance to have to go reading through a large log and this way I am also able to only delete that plugins log.

topic2k commented:
Thanks for adding the split option 😃 
Anything else you want to add, or is it ready to merge?

kdschlosser commented:
I do not think anything else needs to be added. just want to make sure it works 100% because it does have some voodoo code to it. and that is not my favorite kind of code. but it's the only way to do it without adding a method to the plugin base. and then having the plugin devs update their plugins to support it. so i though this would be a better way

topic2k commented:
Now EG only works if i have `splitDebuggingLog = True`. If i set it to `False` or omit it, then EG hangs. After some trying it seems that the logging crashes if the `logs` directory doesn't exists.

kdschlosser committed:
Moves creation of eg.config to before eg.log is created

Had to move eg.Config being initiliazed to before eg.Log due to the log needing an attribute from eg.config. 

Also did some code optimization. and removed the use of os to get the appdata folder.

kdschlosser commented:
i just wanted to let you know this one should be fixed. i forgot to mention it. i did a commit a while back

topic2k commented:
well, sorry to say no. I've deleted Appdata\EventGhost and then run EG with -debug option. The GUI doesn't show up, the process seems to hang and a log file with only `File "C:\Users\topic\AppData\Roaming\EventGhost\config.py" does not exist.` as content is created. After while the EG process exits.
If i run EG without debug option it starts normally.

kdschlosser commented:
after it runs normally if you then close EG and restart it with the -debug parameter does it work then?

topic2k commented:
No, makes no different if i run it successfully before using -debug.
